floodviz/tests/hydrograph_utils_tests.py

```diff
@@ -0,0 +1,106 @@
+
+  def test_bad_list(self):
+    self.assertEqual(req_hydrodata(['x'], self.H_START_DT, self.H_END_DT, self.prefix), None)
+    self.assertEqual(req_hydrodata([123456, 100], self.H_START_DT, self.H_END_DT, self.prefix), None)
+    self.assertEqual(req_hydrodata([], self.H_START_DT, self.H_END_DT, self.prefix), None)
+
+  def test_bad_url(self):
+    self.assertEqual(req_hydrodata(self.sites, self.H_START_DT, self.H_END_DT, 'badurl.noob' ), None)
+    self.assertEqual(req_hydrodata(self.sites, self.H_START_DT, self.H_END_DT, ''), None)
+ 
+
+class TestParseHydroData(unittest.TestCase):
+
+  def setUp(self):
+
+    self.mock_data = [{"sourceInfo": {
+   "siteName": "Black Hawk Creek at Hudson, IA",
+   "siteCode": [{"value": "05463500","network": "NWIS","agencyCode": "USGS"}],
+   "timeZoneInfo": {"defaultTimeZone": {"zoneOffset": "-06:00","zoneAbbreviation": "CST"},"daylightSavingsTimeZone": {
+    "zoneOffset": "-05:00","zoneAbbreviation": "CDT"},"siteUsesDaylightSavingsTime": True},"geoLocation": {
+    "geogLocation": {"srs": "EPSG:4326","latitude": 42.4077639,"longitude": -92.4632451},"localSiteXY": []},
+   "note": [],"siteType": [],},"variable": {"variableCode": [],"variableName": "Streamflow, ft&#179;/s",
+   "variableDescription": "Discharge, cubic feet per second","valueType": "Derived Value","unit": {"unitCode": "ft3/s"},
+   "note": [],},"values": [{"value": [{
+      "value": "446","qualifiers": ["A"],"dateTime": "2008-05-20T00:00:00.000-05:00"},{
+      "value": "446","qualifiers": ["A"],"dateTime": "2008-05-20T00:15:00.000-05:00"},{
+      "value": "446","qualifiers": ["A"],"dateTime": "2008-05-20T00:30:00.000-05:00"},{
+      "value": "444","qualifiers": ["A"],"dateTime": "2008-05-20T00:45:00.000-05:00"},{
+      "value": "446","qualifiers": ["A"],"dateTime": "2008-05-20T01:00:00.000-05:00"},{
+      "value": "446","qualifiers": ["A"],"dateTime": "2008-05-20T01:15:00.000-05:00"}]}]}]
+
+    self.mock_parsed_data = [{'key': 'Black Hawk Creek at Hudson, IA', 'values': [
+     {'date': '2008-05-20', 'time': 0, 'value': 0}, 
+     {'date': '2008-05-20', 'time': '00:00:00', 'value': '446'},
+     {'date': '2008-05-20', 'time': '00:15:00', 'value': '446'},
+     {'date': '2008-05-20', 'time': '00:30:00', 'value': '446'},
+     {'date': '2008-05-20', 'time': '00:45:00', 'value': '444'}, 
+     {'date': '2008-05-20', 'time': '01:00:00', 'value': '446'},
+     {'date': '2008-05-20', 'time': '01:15:00', 'value': '446'}]}] 
+
+
+  def test_empty_data(self):
+    self.assertEqual(parse_hydrodata([]), [])
+    self.assertEqual(parse_hydrodata(None), [])
+
+  def test_data_not_list(self):
+    self.assertEqual(parse_hydrodata({}), [])
+
+  def test_valid_data(self):
+    # Delete 'time_mili' key before comparing due to python time conversion inconsistencies.     
+    ret = parse_hydrodata(self.mock_data)[0]
+    for k1, v1 in ret.items():
```

Comment:
For the future: It might be clearer to do asserts on a value within a dictionary, rather than assertEqual for the whole dictionary. Modifying the result and then comparing just has a bad smell to me.